ProcessList.c

```diff
@@ -663,13 +495,6 @@ void ProcessList_scan(ProcessList* this, bool pauseProcessUpdate) {
    // Set UID column width based on max UID.
    Process_setUidColumnWidth(maxUid);
 
-   if (this->settings->ss->treeView) {
-      // Clear out the hashtable to avoid any left-over processes from previous build
-      //
-      // The sorting algorithm relies on the fact that
-      // len(this->displayTreeSet) == len(this->processes)
-      Hashtable_clear(this->displayTreeSet);
-
+   if (this->settings->ss->treeView)
       ProcessList_buildTree(this);
```

Comment:
Isn't redrawing the panel supposed to sort and rebuild the tree anyway? With this the rebuild is only done for the tree mode, but skipped for the simple sorted list. Can you re-check if simply marking the ProcessList as needs sorting would suffice?

Reply:
It will rebuild the tree, yes, even without marking it here because it's marked as needs sorting in `checkRecalculation` before the call to `ProcessList_scan`. Looks like at the moment there's nothing that would need either tree structure or depth information between scan and redraw.

Reply:
Pushed a commit removing this.